src/Tooltip/Tooltip.tsx

```diff
@@ -1,336 +1,192 @@
-      setTooltipPosition('left')
-      return
-    }
-    // if left out of bounds
-    if (dimensions.left < 0) {
-      setTooltipPosition('right')
-      return
-    }
-    // If bottom out of bounds
-    if (dimensions.bottom > window.innerHeight) {
-      setTooltipPosition('top')
-      return
-    }
-  }, [tipContainer, tooltipPosition])
+  const variantValue = tooltipVariants[variant]
+  const bgColor = bgColorProp ? theme.colors[bgColorProp] : theme.colors.custard
 
-  useEffect(() => {
-    handleTipViewport()
-    setChildEl(document.getElementById(randomId))
-  }, [])
-
-  useEffect(() => {
-    setTooltipPosition(position)
-  }, [position])
-
-  const arrowSize = 18
-
-  const calculateTooltipPosition = (): { top: number; left: number } => {
-    if (!childEl) return { top: 0, left: 0 }
-    if (!tipContainer.current) return { top: 0, left: 0 }
-
-    const childPosition = childEl.getBoundingClientRect()
-    const tipDimensions = tipContainer.current.getBoundingClientRect()
-
-    let top = 0
-    let left = 0
-
-    switch (tooltipPosition) {
-      case 'top':
-        top =
-          childPosition.top + window.scrollY - tipDimensions.height - arrowSize
-        left =
-          childPosition.left +
-          window.scrollX +
-          childPosition.width / 2 -
-          tipDimensions.width / 2
-        break
-      case 'bottom':
-        top = childPosition.bottom + window.scrollY + arrowSize
-        left =
-          childPosition.left +
-          window.scrollX +
-          childPosition.width / 2 -
-          tipDimensions.width / 2
-        break
-      case 'left':
-        left =
-          childPosition.left + window.scrollX - tipDimensions.width - arrowSize
-        top =
-          childPosition.top +
-          window.scrollY +
-          childPosition.height / 2 -
-          tipDimensions.height / 2
-        break
-      case 'right':
-        left = childPosition.right + window.scrollX + arrowSize
-        top =
-          childPosition.top +
-          window.scrollY +
-          childPosition.height / 2 -
-          tipDimensions.height / 2
-        break
-      default:
-        break
-    }
-    return { top, left }
-  }
-
-  useEffect(() => {
-    if (childEl) {
-      setTooltipCoords(calculateTooltipPosition())
-    }
-  }, [
-    position,
-    window.scrollY,
-    tipContainer.current,
-    childEl,
-    childEl?.getBoundingClientRect().top,
-    childEl?.getBoundingClientRect().left,
-  ])
+  const randomId = crypto.randomUUID()
 
-  useEventListener({
-    eventName: 'resize',
-    callback: debounce(handleTipViewport, 100),
-    ref: documentRef,
+  const { refs, floatingStyles, context } = useFloating({
+    open: showTip,
+    placement: position,
+    onOpenChange: setShowTip,
+    middleware: [
+      offset(ARROW_HEIGHT + GAP),
+      flip(),
+      shift(),
+      inlineProp ? inline() : undefined,
+      arrow({ element: arrowRef, padding: 14 }),
+    ],
```

Comment:
[Middleware API](https://floating-ui.com/docs/middleware).

If needed, we can allow for the middleware to be configured at the point of usage.
For now, I'm guessing we don't need that level of configuration.